lean4-infoview/src/infoview/rpcSessions.tsx

```diff
@@ -237,14 +23,28 @@ export function WithRpcSessions({ children }: { children: React.ReactNode }) {
     useClientNotificationEffect(
         'textDocument/didClose',
         (params: DidCloseTextDocumentParams) => {
-            sessions.ensureSessionClosed(params.textDocument.uri)
+            sessions.closeSessionForFile(params.textDocument.uri)
         },
         []
     )
 
-    useEvent(ec.events.serverRestarted, () => sessions.ensureAllSessionsClosed())
+    // TODO: only restart files for the server that stopped
+    useEvent(ec.events.serverRestarted, () => sessions.closeAllSessions())
 
-    return <RpcContext.Provider value={sessions}>
+    return <RpcSessionsContext.Provider value={sessions}>
         {children}
-    </RpcContext.Provider>
+    </RpcSessionsContext.Provider>
+}
+
+const fakeRpcSession: RpcSession =
+    {call: async () => { throw new Error('no rpc context set') }};
+
+export function useRpcSessionTdpp(pos: TextDocumentPositionParams): RpcSession {
+    return React.useContext(RpcSessionsContext)?.connect(pos) || fakeRpcSession;
 }
+
+export function useRpcSessionDp(pos: DocumentPosition): RpcSession {
```

Comment:
`useRpcSessionAtPos` + `useRpcSessionAtTdpp`? I think that's how we usually name `DocumentPosition` vars.